content-services/content-sources-backend
# Build: Use upstream docker image, drop zookeeper

## Summary

This moves us to the official kafka container, as well as drops zookeeper.  Zookeeper is no longer needed, and in fact will removed from the kafka project entirely at 4.0 

## Testing steps

```
make compose-clean compose-up
make run
```

in another terminal
```
KAFKA_TOPICS=platform.content-sources.template make kafka-topic-consume
```

and then create a content template:

```
####
POST http://localhost:8000/api/content-sources/v1.0/templates/
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==
Content-Type: application/json

{
 "name":"FOO",
  "repository_uuids": []
}
```

You should see the message make it to the kafka consume command:

```
Partition:0	content-type:application/cloudevents+json	null	{"specversion":"1.0","id":"ddcf8416-c153-4386-b1c8-9bf338eca641","source":"urn:redhat:source:console:app:repositories","type":"com.redhat.console.repositories.template-created","subject":"urn:redhat:subject:console:rhel:template-created","datacontenttype":"application/json","time":"2024-04-18T11:33:51.38720455Z","data":[{"uuid":"7bdbf4d9-04de-4405-ba4b-1d6505c61477","name":"FOO","org_id":"9","description":"","arch":"","version":"","date":"0001-01-01T00:00:00Z","repository_uuids":[]}],"redhatorgid":"9"}

```

## Checklist
